textinput.go

```diff
@@ -190,4 +200,46 @@ func (p *textinputPlugin) glfwKeyCallback(window *glfw.Window, key glfw.Key, sca
 			}
 		}
 	}
+
+	// key events
+
+	// TODO: Stop using the android keymap and translate the glfw keycode to the
+	// platfom native one
+	// BUG: the LogicalKeyboardKey isn't the right one
+	// https://github.com/flutter/flutter/blob/1f2972c7b6a8503f7c6a5dfa180521a6f7efd472/packages/flutter/lib/src/services/raw_keyboard_android.dart#L116
+
+	// MacOS example: flutter/engine/pull/8219
+	// Linux/Windows Watch: google/flutter-desktop-embedding/issues/323
+	var typeKey string
+	if action == glfw.Release {
+		typeKey = "keyup"
+	} else if action == glfw.Press {
+		typeKey = "keydown"
+	} else {
+		fmt.Printf("go-flutter: failed to send key event, action: %v\n", action)
+		return
+	}
+
+	event := struct {
+		KeyCode   int    `json:"keyCode"`
+		Keymap    string `json:"keymap"`
+		Type      string `json:"type"`
+		MetaState int    `json:"metaState"`
+	}{
+		int(key), "android", typeKey,
+		conditionalInt(mods&glfw.ModShift != 0, androidMetaStateShift) |
```

Comment:
This becomes a bit unreadable due to the positional values and or'ing operation. I suggest to use named fields for these struct values.
(Also, eventually I'd like to generate ffjson code for all structs that we marshal/unmarshall, which greatly improves its performance because no runtime reflection is needed, for that struct types will need to be global in the package)